src/components/atoms/Button/Button.tsx

```diff
@@ -2,148 +2,83 @@ import React, { HTMLAttributes } from 'react';
-
-export type TButtonStylingMapping = { [key in TStyling]: string };
-export type TButtonSizingMapping = { [key in TSizing]: string };
-
-const defaultButtonProps: Partial<IButtonProps> = {
-  sizing: 'default',
-  styling: 'primary',
-};
-
-const smallInset = 'inset 0 0 0 2px';
-const largeInset = 'inset 0 0 0 4px';
-
-const fontColors: Partial<TButtonStylingMapping> = {
-  alert: colors.white,
-  contrastLink: colors.white,
-  contrastSecondary: colors.white,
-  link: colors.actionPlain,
-  primary: colors.white,
-  secondary: colors.greyDark,
-  warning: colors.greyDarkest,
-};
-
-const backgroundColors: TButtonStylingMapping = {
-  alert: colors.alert,
-  contrastLink: 'transparent',
-  contrastPrimary: colors.white,
-  contrastSecondary: 'transparent',
-  link: 'transparent',
-  primary: colors.action,
-  secondary: colors.greyLighter,
-  warning: colors.warning,
-};
-
-const activeBackgroundColors: Partial<TButtonStylingMapping> = {
-  contrastSecondary: 'transparent',
-  secondary: colors.action,
-};
-
-const boxShadows: Partial<TButtonStylingMapping> = {
-  contrastSecondary: `${smallInset} ${colors.white}`,
-  secondary: `${smallInset} ${colors.action}`,
-};
-
-const activeBoxShadows: Partial<TButtonStylingMapping> = {
-  alert: `${smallInset} ${colors.alert}, ${largeInset} ${colors.white}`,
-  contrastPrimary: `${smallInset} ${colors.white}, ${largeInset} ${colors.action}`,
-  contrastSecondary: `${smallInset} ${colors.white}`,
-  primary: `${smallInset} ${colors.action}, ${largeInset} ${colors.white}`,
-  secondary: `${largeInset} ${colors.action}`,
-  warning: `${smallInset} ${colors.warning}, ${largeInset} ${colors.greyDarkest}`,
-};
-
-const hoverFontColors: Partial<TButtonStylingMapping> = {
-  secondary: colors.greyDark,
-};
-
-const fontSizes: TButtonSizingMapping = {
-  compact: '14px',
-  default: '16px',
-  large: '20px',
-  small: '16px',
-};
-
-const paddings: TButtonSizingMapping = {
-  compact: '6px 16px',
-  default: '14px 32px',
-  large: '14px 32px',
-  small: '10px 24px',
+const colorMap = {
+  primary: {
+    text: colors.white,
+    bg: colors.action,
+    hover: `linear-gradient(90deg, #3B46C4 0%, #2732B0 100%)`,
+  },
+  secondary: {
+    text: colors.actionDark,
+    bg: colors.actionLight,
+    hover: shadeHexColor(colors.actionLight, 0.24),
+  },
+  link: {
+    text: colors.actionDark,
+    bg: 'transparent',
+    hover: shadeHexColor(colors.actionLight, 0.24),
+  },
 };
 
-const SText = styled.span<IButtonProps>`
-  display: block;
-  margin-right: ${({ rightIcon }) => !!rightIcon && '8px'};
-  margin-left: ${({ leftIcon }) => !!leftIcon && '8px'};
-`;
-
 export const buttonStyle = css<IButtonProps>`
   text-decoration: none;
   box-sizing: border-box;
   display: inline-flex;
   justify-content: center;
   align-items: center;
   width: ${({ fullWidth = false }) => fullWidth && '100%'};
-  padding: ${({ sizing = 'default' }) => paddings[sizing]};
+  padding: 15px 25px;
```

Comment:
Rob said that we are using a 4px system. This looks a bit weird to me.

Reply:
There weren't specs for the `padding` so I improvised. As you point out better to use the _spacing scale_ here so I'll update 👍🏻